ceremonies/src/lib.rs

```diff
@@ -899,6 +899,63 @@ impl<T: Config> Pallet<T> {
 			.collect()
 	}
 
+	pub fn get_aggregated_account_data(
+		cid: CommunityIdentifier,
+		account: &T::AccountId,
+	) -> AggregatedAccountData<T::AccountId, T::Moment> {
+		let cindex = <encointer_scheduler::Pallet<T>>::current_ceremony_index();
+		let aggregated_account_data_global = AggregatedAccountDataGlobal {
+			ceremony_phase: <encointer_scheduler::Pallet<T>>::current_phase(),
+			ceremony_index: cindex,
+		};
+
+		let aggregated_account_data_personal: Option<
+			AggregatedAccountDataPersonal<T::AccountId, T::Moment>,
+		>;
+
+		// check if the participant is registered. if not the entire personal field will be None.
+		if let Some(participant_type) = Self::get_participant_type((cid, cindex), account) {
+			let mut meetup_location_index: Option<MeetupIndexType> = None;
+			let mut meetup_time: Option<T::Moment> = None;
+			let mut meetup_registry: Option<Vec<T::AccountId>> = None;
+			let mut meetup_index: Option<MeetupIndexType> = None;
+
```

Comment:
Why do you predefine those fields, we same some lines of code if this is defined in place.

Reply:
i do it because i use a bunch of if statements to assign those variables. if i dont predefine them here, they woule be undefined in some cases. if there is a more elegant solution to this, please give me a hint :)